packages/azure-kusto-data/src/tokenProvider.ts

```diff
@@ -314,125 +254,77 @@ export class UserPassTokenProvider extends MsalTokenProvider {
-        return this.msalClient.acquireTokenByClientCredential({
-            scopes: this.scopes,
-        });
+        );
     }
 
-    context(): Record<string, any> {
-        return { ...super.context(), clientId: this.appClientId };
-    }
 }
 
 /**
  * Acquire a token from MSAL using application certificate
  * Passing the public certificate is optional and will result in Subject Name & Issuer Authentication
  */
-export class ApplicationCertificateTokenProvider extends MsalTokenProvider {
-    msalClient!: ConfidentialClientApplication;
-
-    constructor(
+export class ApplicationCertificateTokenProvider extends AzureIdentityProvider {
+    constructor (
         kustoUri: string,
         private appClientId: string,
-        private certThumbprint: string,
         private certPrivateKey: string,
-        private certX5c?: string,
+        private sendX5c?: boolean,
         authorityId?: string
     ) {
-        super(kustoUri, authorityId!, appClientId);
+        super(kustoUri, authorityId!);
     }
 
-    initClient(): void {
-        const commonOptions = this.commonOptions();
-        const clientConfig = {
-            ...commonOptions,
-            auth: {
-                ...commonOptions.auth,
-                clientCertificate: {
-                    thumbprint: this.certThumbprint,
-                    privateKey: this.certPrivateKey,
-                    x5c: this.certX5c,
-                },
-            },
-        };
-        this.msalClient = new ConfidentialClientApplication(clientConfig);
+    getCredential(): TokenCredential {
+        return new ClientCertificateCredential(
+            this.authorityId!,
+            this.appClientId!,
+            {
+                certificate: this.certPrivateKey
+            } as ClientCertificatePEMCertificate,
+            {
+                sendCertificateChain: this.sendX5c
+            } as ClientCertificateCredentialOptions,
+        );
     }
+}
 
-    acquireTokenWithCloudSettings(): Promise<TokenType | null> {
-        return this.msalClient.acquireTokenByClientCredential({
-            scopes: this.scopes,
-        });
+
+/**
+ * Acquire a token from MSAL with application id and Key
+ * For browser cors: you need to visit your app registration and update the redirect URI you're using to the type spa (for "single page application").
+ * TODO: currently failing
+ */
+ export class ApplicationKeyTokenProvider extends AzureIdentityProvider {
+    constructor(kustoUri: string, private appClientId: string, private appKey: string, authorityId: string) {
+        super(kustoUri, authorityId);
+    }
+
+    getCredential(): TokenCredential {
+        return new ClientSecretCredential(
+            this.authorityId!, // The tenant ID in Azure Active Directory
+            this.appClientId, // The app registration client Id in the AAD tenant
+            this.appKey // The app registration secret for the registered application
```

Comment:
SPA can't really use an app secret right? Since it would require the client browser to know said secret